test.js

```diff
@@ -311,6 +311,19 @@ test('the callback should provide the original proxied object, the path to the c
 	t.deepEqual(returnedPrevious, [{z: 4}, {z: 3}, {z: 2}]);
 	t.deepEqual(returnedValue, [{z: 4}]);
 	t.is(callCount, 12);
+
+	proxy.foo = function () {
+		proxy.x.y[0].z = 2;
+		t.is(returnedObject, proxy);
+		// Fails here. returnedPath === 'foo' instead of 'x.y.0.z'
```

Comment:
The returned path here is foo because the last change made was adding the foo method to proxy. on-change intentionally catches all changes that happen within a proxied method call and only calls the onChange callback if something changed during the method call.